app/components/page-components/Chatbot/windows/ChatbotSongRequestPreferencesWindow.vue

```diff
@@ -0,0 +1,136 @@
+
+<template>
+<ModalLayout
+  :showControls="false"
+  :customControls="true"
+  :title="$t('Song Request Preferences')"
+>
+  <div slot="fixed">
+    <div class="row">
+      <div class="small-6 columns position--relative window-tab">
+        <Tabs :tabs="tabs" :value="selectedTab" @input="onSelectTabHandler"></Tabs>
+      </div>
+      <div class="small-6 columns position--relative window-tab">
+        <div class="window-toggle__wrapper">
+          <div @click="onToggleSongRequestWindowHandler">
+            <span> {{ $t('Edit Primary Command') }} </span>
+            <i class="fas fa-chevron-right window-toggle__icon"></i>
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+  <div slot="content" class="chatbot-song-request__container">
+    <div v-if="songRequestData">
+      <transition name='fade' mode="out-in" appear>
+        <div v-if="selectedTab === 'general' && !!songRequestData.general">
+          <VFormGroup
+            :title="$t('Max Duration (Value in Seconds)')"
+            v-model="songRequestData.general.max_duration"
+            :metadata="metadata.general.max_duration"
+          />
+          <VFormGroup
+            :title="$t('Spam Security')"
+            v-model="songRequestData.general.filter_level"
+            :metadata="metadata.general.filter_level"
+          />
+        </div>
+        <div v-else>
+          <table v-if="songRequestBannedMedia.length > 0">
+            <thead>
+              <tr>
+                <th> {{ $t('Video') }} </th>
+                <th> {{ $t('Banned By') }} </th>
+                <th> {{ $t('Unban') }} </th>
+              </tr>
+            </thead>
+            <tbody>
+              <tr
+                v-for="media in songRequestBannedMedia"
+                :key="media.id"
+              >
+                <td> {{ media.media_title }} </td>
+                <td> {{ media.action_by }} </td>
+                <td>
+                  <button
+                    @click="onUnbanMediaHandler(media)"
+                    class="button button--default"
+                  >{{ $t('Unban') }}</button>
+                </td>
+              </tr>
+            </tbody>
+          </table>
+          <div v-else class="chatbot-empty-placeholder__container">
+            <img
+              :src="require(`../../../../../media/images/chatbot/chatbot-placeholder-blacklist--${this.nightMode ? 'night' : 'day'}.svg`)"
+              width="200"
+            />
+            {{ $t('No items in list. Add new.') }}
+          </div>
+        </div>
+      </transition>
+    </div>
+  </div>
+  <div slot="controls" class="flex flex--space-between">
+    <div>
+      <button
+        class="button button--default"
+        @click="onCancelHandler">
+        {{ $t('Cancel') }}
+      </button>
+      <button
+        class="button button--action"
+        @click="onSaveHandler"
+        :disabled="errors.items.length > 0"
```

Comment:
since sometimes user can't see why this button is disabled it's better to add a validation in `onSaveHandler` as I refactored in other places